tests/core/test_blob_utils.py

```diff
@@ -22,25 +27,38 @@
 )
 
 
+BLOB_SERIALIZATION_TEST_DATA = [  # [(blobs, unpadded_body), ...]
+    ([], b""),
+    ([b"\x00"], b"\x01\x00"),
+    ([b"\x01"], b"\x01\x01"),
+    ([b"\xaa" * 31], b"\x1f" + b"\xaa" * 31),
+    ([b"\xaa" * 32], b"\x00" + b"\xaa" * 31 + b"\x01" + b"\xaa"),
+    (
+        [b"\xaa" * 20, b"\xbb" * 15],
+        b"\x14" + b"\xaa" * 20 + b"\x00" * 11 + b"\x0f" + b"\xbb" * 15
```

Comment:
These test cases are really opaque.  Maybe you could split these up across multiple lines like this.

```python
(
    [
        b"\xaa" * 20,  # first blob
        b"\xbb" * 15,  # second blob
    ],
    (
        b"\x14"  # whatever this value is
        b"\xaa * 20"  # the first blob body
        b"\x00 * 11"  # padding
        b"\x0f"  # whatever this value is
        b"\xbb" * 15  # the second blob body
    )
)
```